pocs/utils/database.py

```diff
@@ -101,14 +103,46 @@ def insert_current(self, collection, obj, include_collection=True):
             }
 
             # Update `current` record
-            self.current.replace_one({'type': collection}, current_obj, True)
+            _id = self.current.replace_one(
+                {'type': collection}, current_obj, True  # True for upsert
+            ).upserted_id
 
             if include_collection:
-                # Insert record into db
-                col = getattr(self, collection)
-                _id = col.insert_one(current_obj).inserted_id
-        except AttributeError:
-            warn("Collection does not exist in db: {}".format(collection))
+                _id = self.insert(collection, current_obj)
+        except Exception as e:
+            warn("Problem inserting object into collection: {}".format(e))
+
+        return _id
+
+    def insert(self, collection, obj):
+        """Insert an object into the collection provided.
+
+        Args:
+            collection (str): Name of valid collection within panoptes db.
+            obj (dict or str): Object to be inserted.
+
+        Returns:
+            str: Mongo object ID of record in `collection`.
+        """
+        assert collection in self.collections, warn("Collection not available")
+
+        _id = None
+        try:
+            # If `data` key is present we assume it has "metadata" for object
+            if 'data' in obj:
+                # But still check for a `type`
+                if 'type' not in obj:
+                    obj['type'] = collection
+            else:
+                obj = {
+                    'type': collection,
+                    'data': obj,
+                    'date': current_time(datetime=True),
+                }
+
+            # Insert record into db
+            col = getattr(self, collection)
+            _id = col.insert_one(obj).inserted_id
         except Exception as e:
             warn("Problem inserting object into collection: {}".format(e))
```

Comment:
Could you add an additional message with the obj to be inserted?
And note that this is not written to the log file, which would really be useful!

Reply:
Fixed this so now we have a logger in PanMongo and warnings will go to appropriate place.